inc/umock_c/umock_c_prod.h

```diff
@@ -54,6 +57,31 @@
 #define MOCKABLE_FUNCTION_WITH_CODE_END() \
     MOCK_FUNCTION_END()
 
+/* Codes_SRS_UMOCK_C_LIB_01_217: [ In the presence of the `ENABLE_MOCKS` define, `IMPLEMENT_MOCKABLE_FUNCTION` shall expand to the signature of the function, but the name shall be changed to be prefix with `real_`. ]*/
+#define IMPLEMENT_MOCKABLE_FUNCTION(modifiers, result, function, ...) \
+    MOCKABLE_FUNCTION_SIGNATURE(modifiers, result, MU_C2(real_, function), __VA_ARGS__)
+
+#define EXTRACT_ARGS_FROM_FUNCTION(modifiers, result, function, ...) \
+    modifiers, result, function, __VA_ARGS__
+
+#define EXPAND_ENTRY(A) \
+    MOCKABLE_##A
+
+#define REGISTER_GLOBAL_MOCK_REAL_FUNCTION(modifiers, result, function, ...) \
+    REGISTER_GLOBAL_MOCK_HOOK(function, MU_C2(real_, function));
+
+#define REGISTER_GLOBAL_MOCK_REAL(A) \
+    REGISTER_GLOBAL_MOCK_REAL_##A
+
+/* Codes_SRS_UMOCK_C_LIB_01_215: [ Each item in `...` shall be an entry for one mockable function. ]*/
+/* Codes_SRS_UMOCK_C_LIB_01_216: [ Each item in `...` shall be defined using a macro called `FUNCTION`, which shall be an alias for `MOCKABLE_FUNCTION`. ]*/
+#define MOCKABLE_INTERFACE(interface_name, ...) \
+    MU_FOR_EACH_1(EXPAND_ENTRY, __VA_ARGS__) \
```

Comment:
>EXPAND_ENTRY [](start = 18, length = 12)

Alright, now I get why there was a FUNCTION(....) in the .md. Can we phlease have MOCKABLE_FUNCTION, and not just FUNCTION? Would keep it easy and the upgrade path would be much simplified. . .

Reply:
Will try. Let's see if I can make it compile :-)

---
In reply to: [293557071](https://github.com/Azure/umock-c/pull/95#discussion_r293557071) [](ancestors = 293557071)

Reply:
Humpf, sorry. Maybe some other day. Too hard to control when things get expanded.

---
In reply to: [293637886](https://github.com/Azure/umock-c/pull/95#discussion_r293637886) [](ancestors = 293637886,293557071)